tests/unit/model-test.js

```diff
@@ -681,6 +681,26 @@ module('unit/model', function(hooks) {
     assert.equal(get(model, 'publisher'), 'Harper Collins, of course', 'specified value transformed');
   });
 
+  test('early set of an ID to a newly created records is allowed', function(assert) {
```

Comment:
potentially I should just make this a different PR and go ahead and land it and the implementation into master. cc @hjdivad

Reply:
did so here https://github.com/hjdivad/ember-m3/pull/53